pallets/erc20-xcm-bridge/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Pallet that allow to transact erc20 tokens trought xcm directly.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+mod erc20_matcher;
+mod erc20_trap;
+mod errors;
+mod xcm_holding_ext;
+
+use frame_support::pallet;
+
+pub use erc20_trap::AssetTrapWrapper;
+pub use pallet::*;
+pub use xcm_holding_ext::XcmExecutorWrapper;
+
+#[pallet]
+pub mod pallet {
+
+	use crate::erc20_matcher::*;
+	use crate::errors::*;
+	use crate::xcm_holding_ext::*;
+	use ethereum_types::BigEndianHash;
+	use fp_evm::{ExitReason, ExitSucceed};
+	use frame_support::pallet_prelude::*;
+	use pallet_evm::Runner;
+	use sp_core::{H160, H256, U256};
+	use sp_std::vec::Vec;
+	use xcm::latest::{Error as XcmError, MultiAsset, MultiLocation, Result as XcmResult};
+	use xcm_executor::traits::{Convert, Error as MatchError, MatchesFungibles};
+	use xcm_executor::Assets;
+
+	const ERC20_TRANSFER_CALL_DATA_SIZE: usize = 4 + 32 + 32; // selector + from + amount
+	const ERC20_TRANSFER_GAS_LIMIT: u64 = 200_000;
```

Comment:
why not as `Get<u64>` in the pallet config?

Reply:
Done